src/main.cpp

```diff
@@ -48,6 +53,41 @@ int runMixxx(MixxxApplication* pApp, const CmdlineArgs& args) {
     }
 }
 
+void adjustScaleFactor(CmdlineArgs* pArgs) {
+    if (qEnvironmentVariableIsSet(kScaleFactorEnvVar)) {
+        bool ok;
+        const double f = qgetenv(kScaleFactorEnvVar).toDouble(&ok);
+        if (ok && f > 0) {
+            // The environment variable overrides the preferences option
+            qDebug() << "Using" << kScaleFactorEnvVar << f;
+            pArgs->storeScaleFactor(f);
+            return;
+        }
+    }
+    // We cannot use ConfigObject, because it depends on MixxxApplication
+    // but the scale factor is read during it's constructor.
+    // QHighDpiScaling can not be used afterwards because it is private.
+    auto cfgFile = QFile(QDir(pArgs->getSettingsPath()).filePath(MIXXX_SETTINGS_FILE));
+    if (cfgFile.open(QFile::ReadOnly | QFile::Text)) {
+        QTextStream in(&cfgFile);
+        QString strScaleFactor;
+        QString line = in.readLine();
+        while (!line.isNull()) {
+            if (line.startsWith(kScaleFactorConfigKey)) {
```

Comment:
Have you considered using `QSettings` instead of hacking a new config parser here?

Reply:
Yes, unfortunately the attempt failed because QSettings expects a "=" between key and value, which is not used in our ini file.

Reply:
Ok. It looks like only the constructor depends on QApplication to construct file paths. Can you just add a ConfigObject static method like `ConfigObject::fromFilePath(path)`?